staging/src/k8s.io/client-go/plugin/pkg/client/auth/exec/metrics.go

```diff
@@ -17,10 +17,25 @@ limitations under the License.
 package exec
 
 import (
+	"os/exec"
 	"sync"
 	"time"
 
 	"k8s.io/client-go/tools/metrics"
+	"k8s.io/klog/v2"
+)
+
+// The following constants shadow the special values used in the prometheus metrics implementation.
+const (
+	// pluginSuccessExitCode is the exit code passed to the metrics.ExecPluginCalls counter when a
+	// call to an exec plugin succeeds.
+	pluginSuccessExitCode = 0
+	// pluginNotFoundExitCode is the exit code passed to the metrics.ExecPluginCalls counter when a
+	// call to an exec plugin fails because the exec plugin is not found.
+	pluginNotFoundExitCode = -1
+	// unrecognizedPluginExitCode is the exit code passed to the metrics.ExecPluginCalls counter when
+	// a call to an exec plugin returns an error that the code does not understand.
+	unrecognizedPluginExitCode = -2
```

Comment:
this seems overly clever to me... admission webhooks have a similar issue where an API call can be rejected in admission because 1) there was an API server error, 2) there was an error invoking the webhook, or 3) the admission webhook was called and correctly rejected the request.

They represent those cases with an extra label in the webhook_rejection_count metric (error_type = calling_webhook_error | apiserver_internal_error | no_error), rather than try to overload the code label

Reply:
thanks @liggitt - i took a pass at this in e0b334b4204d62a3e7b13d6c20e15c226438624c

i tried to take the concept you shared and adapt it to this particular metric that we are mucking with. hopefully i got it somewhere close to where we want it.